.github/workflows/check-jax-release.yaml

```diff
@@ -30,54 +25,70 @@ jobs:
     - name: Install deps
       run: |
         sudo apt-get install -y cmake ninja-build clang lld libomp-dev
+        mkdir -pv _venv
+        python3 -m venv _venv
+        . _venv/bin/activate
```

Comment:
Why the virtual environment?

Reply:
Hmm, good question. I checked against the build-wheel workflow and I thought that the answer is because this workflow runs on manylinux image which I believe runs as root (I can see [these lines](https://github.com/PennyLaneAI/catalyst/blob/8415c6130813c7c3c9b547a9680201e79897a112/.github/workflows/build-wheel-linux-x86_64.yaml#L193) work). But your other examples are interesting. My guess is that it is because they don't do editable installs as we actually do in [make frontend](https://github.com/PennyLaneAI/catalyst/blob/8415c6130813c7c3c9b547a9680201e79897a112/Makefile#L40). See also the error log I had before I added venv https://github.com/PennyLaneAI/catalyst/actions/runs/6498419351/job/17649722722

Reply:
@maliasadi could you remember what problems did we have with `pip install --user` flag?

Reply:
@dime10 I was able reproduce the issue using a local ubuntu-latest [docker container](https://github.com/XanaduAI/sergei-environment/blob/803a7701a837992af00bb09a8e59bb3f9d1fb428/docker/dev_novenv.dockerfile#L1). I can see the following:
* read-only site-packages is not really an issue because `pip` detects it and switches into the user-mode automatically.
* But editable installs do not work if the default site-packages is read-only. All your examples do not use `-e` flag, so they work. Our Makefile command includes `-e` and fails unless we take precautions.

Reply:
For reference, it seems editable installs require `pip>=23` to work, due to some bug in the install process where it tests that the system site-packages are writable even if the user directory is selected for installation.

I think we could do two things:
- document that it is required in our build from source instructions
- we don't actually need editable installs for these actions, so we could disable it here like we for the actions (I'm thinking we could also add a flag to the makefile if we want to keep using `make frontend`)

Reply:
So I have added a newer `pip` dependency into the requirements, now it works without venv.